src/bucket.ts

```diff
@@ -258,6 +273,7 @@ export class S3Bucket {
       versionId: res.headers.get("x-amz-version-id") ?? undefined,
       websiteRedirectLocation:
         res.headers.get("x-amz-website-redirect-location") ?? undefined,
+      meta: Object.keys(meta).length !== 0 ? meta : undefined,
```

Comment:
change this to always be set to meta (non optional). If no meta keys are set, this should be an empty object.